rclcpp/src/rclcpp/executor.cpp

```diff
@@ -304,9 +305,13 @@ Executor::execute_any_executable(AnyExecutable & any_exec)
   }
   // Reset the callback_group, regardless of type
   any_exec.callback_group->can_be_taken_from().store(true);
+
+  // Check whether triggering a guard condition is necessary
+  // (will depend on the type of executor and callback groups)
   // Wake the wait, because it may need to be recalculated or work that
   // was previously blocked is now available.
-  if (rcl_trigger_guard_condition(&interrupt_guard_condition_) != RCL_RET_OK) {
+  if (set_wake_after_execute_flag() &&
+      rcl_trigger_guard_condition(&interrupt_guard_condition_) != RCL_RET_OK) {
```

Comment:
nitpick: If you have a multi-line if statement, put the `{` on its own line:

```suggestion
  if (set_wake_after_execute_flag() &&
      rcl_trigger_guard_condition(&interrupt_guard_condition_) != RCL_RET_OK)
  {
```